src/data-fetchers/vcf/vcf-worker.ts

```diff
@@ -184,10 +184,14 @@ const tile = async (uid: string, z: number, x: number): Promise<void[]> => {
                 DISTPREVLOGE
             };
 
+            // Add optional INFO columns
             Object.keys(vcfRecord.INFO).forEach(key => {
                 const val = vcfRecord.INFO[key];
-                if (Array.isArray(val)) return [key, val[0]] as const;
-                return [key, val] as const;
+                if (Array.isArray(val)) {
+                    data[key] = val.join(', ');
+                } else {
+                    data[key] = val;
+                }
```

Comment:
This closure has several responsibilities:

1.) parse a `VcfRecord` from a string
2.) convert a `VcfRecord` to `VcfTile`
3.) add the result to a global cache


Let's move the logic for 2 to a separate function, e.g., `recordToTile`. All this custom logic for converting a record to a tile can live in a function (which we could also unit test).

```typescript
       const parseLineStoreData = (line: string, prevPos?: number) => {
            const vcfRecord: VcfRecord = parser.parseLine(line);
            const data = recordToTile(vcfRecord, prevPos);

            // Store this column
            tileValues[CACHE_KEY] = tileValues[CACHE_KEY].concat([data]);

            // Return current POS
            return vcfRecord.POS;
        };
```

Reply:
Thank you for this suggestion. I made this change and also added a unit test.